### ISSUE TITLE
Notations test suite is too weak

### ISSUE THREAD

**Issue by maximedenes** [coq/coq#7002]
I think that bugs introduced by changes like #873 and #6651 should have been caught by the notation test suite. That test suite should be reorganized (to follow the documentation) and made more complete.

**Comment by herbelin** [comment]
The change of semantics with respect to abbreviations in #873 was observed in the test suite (and the test suite changed accordingly in #873). The question was not to detect it, the question was about how to address it.

The inefficiency in #6952 was not detected but that would be hard to predict in advance all situations where using an efficient algorithm crucially matters. Do you have a proposal in mind?

Regarding #6651, as far as I understood it is also related to a design choice to do wrt primitive projections. By the way, didn't you promise to revert #6651?

**Comment by maximedenes** [comment]
> By the way, didn't you promise to revert #6651?

Yes but I'll branch first in order to push the revert only to the 8.8 branch. I didn't do that for the other revert since I didn't want to do it at the last minute (it was a bit trickier to fix the test suite), and didn't want to backport all the merges of the Sphinx chapters.

**Comment by herbelin** [comment]
@maximedenes: OK (and I shall not ask you about the design decisions to take, since I guess you have other urgencies as of today than thinking at design decisions).
